pallets/committee/src/lib.rs

```diff
@@ -348,33 +363,45 @@ pub mod pallet {
 		}
 
 		/// Extrinsic to vote on an existing proposal.
-		/// This can only be called by members of the committee.
-		/// Successfully cast votes will be recorded in the state and a proposal
-		/// meeting voting requirements can be executed.
+		///
+		/// This can only be called by members of the committee that are eligible to vote.
+		///
+		/// New members are eligible to vote after 1 voting period has passed from the block they
+		/// were added to the members set. Successfully cast votes will be recorded in the state and
+		/// a proposal meeting voting requirements can be executed.
+		#[transactional]
 		#[pallet::weight((T::WeightInfo::vote(), DispatchClass::Operational))]
 		pub fn vote(origin: OriginFor<T>, proposal_hash: HashFor<T>, vote: VoteKind) -> DispatchResult {
-			// Only members can vote
 			let voter = Self::ensure_member(origin)?;
 
-			Votes::<T>::try_mutate(&proposal_hash, |votes| {
-				if let Some(votes) = votes {
-					// Can only vote within the allowed range of blocks for this proposal
-					ensure!(Self::within_voting_period(votes), Error::<T>::NotInVotingPeriod);
-					// members can vote only once
-					ensure!(!votes.has_voted(&voter.account_id), Error::<T>::DuplicateVote);
-					votes.cast_vote(MemberVote::new(voter.clone(), vote.clone())); // mutates votes in place
-					Self::deposit_event(Event::VoteCast(voter, proposal_hash, vote));
-					Ok(())
-				} else {
-					Err(Error::<T>::NoProposalWithHash)
-				}
-			})?;
+			VotingEligibility::<T>::get(&voter.account_id)
+				.and_then(|block_number| {
+					if frame_system::Pallet::<T>::block_number() >= block_number {
+						Some(block_number)
+					} else {
+						None
+					}
+				})
+				.ok_or(Error::<T>::NotEligibileToVoteYet)?;
```

Comment:
can we make this:

`.filter(|block_number|frame_system::Pallet::<T>::block_number() >= *block_number)`

instead

Reply:
forgot that recording the eligibilities now